
##########################################################################
##                             TOWER SETTINGS                           ##
##########################################################################

num_towers: 4                   # number of towers
num_charge_leds_per_tower: 4    # number of charging leds per tower
charging_time: 2.5              # charging time for a led
led_feedback_duration: 0.25      # blink duration for the tower status led when
                                #  feedbacking button presses

tilt_sensor_topic_name: tower/tilt_sensor
button_topic_name: tower/button_state
behavior_service_name: change_behavior

# XY position of towers (SHOULD BE UPDATEd FOR EACH NEW GAME ENVIRONMENT):

# for AIRlab
tower_1: [-0.375873476267, -1.73176705837] 
tower_2: [-0.0357848182321, 1.63595640659]
tower_3: [3.42944264412, 1.23735296726]
tower_4: [3.02523159981, -2.09340715408]

##########################################################################
##                         DIFFICULTY SETTINGS                          ##
##########################################################################

# Here, we assume there exist some set of difficulties. For each one of
# of those, we set "max_speed" (both x and y axis), "min_speed", and the
# exponent of the "blocking factor".
 
difficulties: {
                            
                easy: {max_speed: 0.5,
                       min_speed: 0.05,
                       max_acc: 0.1,
                       ks: 50., 
                       bf_exponent: 1},
                       
                medium: {max_speed: 0.7,
                         min_speed: 0.5,
                         max_acc: 0.5,
                         ks: 50., 
                         bf_exponent: 2},
                         
                       
                hard: {max_speed: 1.,
                       min_speed: 0.7,
                       max_acc: 0.1,
                       ks: 50., 
                       bf_exponent: 1}
               }

current_difficulty: medium

block_timeout: 3.0    #the time span for consider the player has blocked the robot
